src/Core/src/Core/IIndicatorView.cs

```diff
@@ -0,0 +1,28 @@
+﻿using Microsoft.Maui.Graphics;
+
+namespace Microsoft.Maui
+{
+	public interface IIndicatorView : IView
+	{
+		int Count { get; }
+
+		int Position { get; set; }
+
+		double IndicatorSize { get; }
+
+		int MaximumVisible { get; }
+
+		bool HideSingle { get; }
+
+		Paint? IndicatorColor { get; }
+
+		Paint? SelectedIndicatorColor { get; }
+
+		IShape IndicatorsShape { get; }
```

Comment:
We use _IShape_ to determine whether to use shapes or the template, but IShape could be anything (polygon, path etc) while we only support Ellipse and Rectangle. We could use an interface like `IIndicatorShape` and implement it only in the Ellipse and Rectangle classes.

Reply:
We dont use IShape we use the ITemplatedIndicatorView to decide if we need to use IndicatorsShape or the templated